WeScan/Scan/CaptureSessionManager.swift

```diff
@@ -62,8 +62,13 @@ final class CaptureSessionManager: NSObject, AVCaptureVideoDataOutputSampleBuffe
     
     // MARK: Life Cycle
     
-    init?(videoPreviewLayer: AVCaptureVideoPreviewLayer) {
+    init?(videoPreviewLayer: AVCaptureVideoPreviewLayer, delegate: RectangleDetectionDelegateProtocol? = nil) {
```

Comment:
Would it make sense to make the delegate not optional at this point?
Using the `CaptureSessionManager` without a delegate doesn't really make sense.

Reply:
That's an interesting question, Boris!

Obviously you know the code better than I do, so if you're saying that setting the delegate is critical then, yes, we should go ahead and make it non-optional and make sure it's set at init.

I made it optional because previously `CaptureSessionManager` could have been initialized and used without the `captureSessionManager?.delegate = self` line, so I thought it best to preserve that possibility.

A few alternate solutions that spring to mind are:

1. Since the delegate needs to be set, if it is set, at initialization, the user should be forced to set it there only --- we could make `weak var delegate` private(set) access so that it cannot be set anywhere other than initialization (that would prevent any other occurrences in the code of setting the delegate after init). This solution would be improved by removing the default value of `nil` from the init argument and updating the `CaptureSessionManager` summary to explain what's going on here.

2. Another option would be to remove the error handling by the delegate inside the initializer completely and throw the errors instead. This might be a good solution because then the delegate assignments throughout the code don't have to change at all, and being able to set the delegate after initialization is the expected behavior that everyone's used to.

Reply:
I changed this so the delegate is non-optional in the init, and made the delegate private so that it can't be made nil after init.

Reply:
Great, thank you @davidsteppenbeck! 
> I made it optional because previously CaptureSessionManager could have been initialized and used without the captureSessionManager?.delegate = self line, so I thought it best to preserve that possibility.

That makes total sense.

I like this solution you implemented.